core/http/src/method.rs

```diff
@@ -47,7 +47,9 @@ pub enum Method {
     /// The `CONNECT` variant.
     Connect,
     /// The `PATCH` variant.
-    Patch
+    Patch,
+    /// A wildcard variant meant to accept any method
+    Any = u8::MAX,
```

Comment:
This breaks the `Method` enum in that `Any` is not actually a method. We need to find another way to represent "any". My recommendation is to use an `Option<Method>` for the route's method, where `Some(Method)` indicates it matches a (single) method, and `None` indicates it matches all methods.